mesonbuild/mesonlib.py

```diff
@@ -1080,6 +1080,10 @@ def Popen_safe(args: T.List[str], write: T.Optional[str] = None,
                **kwargs: T.Any) -> T.Tuple[subprocess.Popen, str, str]:
     import locale
     encoding = locale.getpreferredencoding()
+    # Redirect stdin to DEVNULL otherwise the command run by us here might mess
+    # up the console and ANSI colors will stop working on Windows.
+    if 'stdin' not in kwargs:
+        kwargs['stdin'] = subprocess.DEVNULL
```

Comment:
We have many more places where we use Popen and friends directly instead of this wrapper, I think. Should we take this opportunity to convert them all to use a meson wrapper function? I remember you did the same change for git invokations in wrap.py recently.

Reply:
btw, for git you also added that hack:

```python
    if platform.system().lower() == 'windows':
        mlog._windows_ansi()
```

Sounds like something we should write once, and use everywhere.

Reply:
Shouldn't that just happen in the mlog module at the top scope?

Reply:
I was trying to not call `mlog._windows_ansi()` on every subprocess invocation since that's already what we spend the most amount of time on during configure.. I'll measure it and get some actual numbers.

Closing stdin is good hygiene for `run_command()` anyway to avoid the program expecting something on stdin and blocking configure.

The mlog module only calls `_windows_ansi()` only once, while the problem here is that the console mode gets reset by subprocesses sometimes, and the way to detect that also involves setting it, so we can just set it.

Reply:
On both my old Windows machine and my new Windows machine, calling `mlog._windows_ansi()`  100,000 times in a loop takes ~2-3 seconds. That's basically negligible. I'll call it on every subprocess invocation.